File: crates/cs/csharp/ZeroDDS.Cdr/src/EndianMode.cs
// SPDX-License-Identifier: Apache-2.0
// Copyright 2026 ZeroDDS Contributors
//
// XCDR2 Endian-Mode-Enum.
// Spec: zerodds-xcdr2-csharp-1.0 §2 + zerodds-xcdr2-bindings-conformance-1.0 §3.

namespace ZeroDDS.Cdr;

/// <summary>
/// Endianness selection for the XCDR2 encoder/decoder.
/// PLAIN_CDR2 LE is the default wire format per Conformance spec §3
/// (`0x00 0x01 0x00 0x00`-Encoding-Header).
/// </summary>
public enum EndianMode
{
    /// <summary>Little-endian (default for PLAIN_CDR2).</summary>
    LittleEndian,

    /// <summary>Big-endian (for the key hash `PlainCdr2BeKeyHolder`, XTypes §7.6.8).</summary>
    BigEndian,
}

File: crates/java-omgdds/java/src/main/java/org/zerodds/cdr/EndianMode.java
// SPDX-License-Identifier: Apache-2.0
package org.zerodds.cdr;

/**
 * Endianness for XCDR2 encoding.
 *
 * <p>Spec zerodds-xcdr2-bindings-conformance-1.0 §3: Default ist
 * {@code LITTLE_ENDIAN} (PLAIN_CDR2 LE Encapsulation
 * {@code 0x00 0x01 0x00 0x00}). {@code BIG_ENDIAN} for key-hash
 * PlainCdr2BeKeyHolder (XTypes §7.6.8).
 */
public enum EndianMode {
    /** Little-endian (default for wire encoding). */
    LITTLE_ENDIAN,
    /** Big-endian (for key hash computation). */
    BIG_ENDIAN
}
